### PR TITLE
[train] Disable `local_shuffle_buffer_size` in train data ingest benchmark

### PR DESCRIPTION
**Pull request by justinvyu**
## Summary

`local_shuffle_buffer_size` is known to be a bottleneck, so for now, we'll exclude this from the benchmark and tackle the other bottlenecks in data ingestion.

`local_shuffle_buffer_size` is not scalable by its current design, since it happens on the iterator side, which is just the train worker. We will consider adding a distributed version of the shuffle buffer in the future (that's in-between no shuffling and a global shuffle).

This PR also adds some more config utilities to the release test fixture. You can now add nested configs in the `BenchmarkConfig` and also dynamically choosing which configs are available as CLI arguments depending on certain top-level configs.
* For example, `dataloader_type=RAY_DATA` enables all configs in `RayDataConfig` to be accessible.

## Benchmark Results

See https://github.com/ray-project/ray/pull/50019 for the original results WITH `local_shuffle_buffer_size=8*32 (8x batch size)`. 

### `variant=.skip_training`

| Metric                         | Value                      |
|---------------------------------|----------------------------|
| **train/global_throughput**    | 7478.26 rows/s             |
| train/local_throughput         | 467.39 rows/s              |
| producer_throughput            | 8049.64 rows/s             |
| block_to_batch-avg             | 0.001 s                   |
| collate-avg                    | 0.0078 s                   |
| fetch_block-avg                | 0.2508 s                   |
| format_batch-avg               | 0.02469 s                   |
| train/iter_batch-total | 159.16 |
| train/iter_first_batch-total | 12.1381 s |
| train/step-total | 0 s |

[See here for the full set of raw metrics.](https://buildkite.com/ray-project/release/builds/32172#0194dfbd-a086-4b2f-ad25-8a6e2ad3b188)

### `variant=.full_training`

Here’s the markdown table summarizing the relevant training metrics:

| Metric                        | Value                      |
|--------------------------------|----------------------------|
| **train/global_throughput**       | 1639.28 rows/s                   |
| train/local_throughput        | 102.45 rows/s                    |
| producer_throughput           | 1983.55 rows/s                    |
| fetch_block-avg               | 0.01149 s                    |
| block_to_batch-avg            | 0.00076 s                     |
| format_batch-avg              | 0.01796 s                    |
| collate-avg                   | 0.2569 s                     |
| train/iter_batch-total | 0.1710 s |
| train/iter_first_batch-total | 12.60 s |
| train/step-total | 768.76 s |

[See here for the full set of raw metrics.](https://buildkite.com/ray-project/release/builds/32172#0194dfbd-a085-443d-b764-b679678f64a4)

### CODE REVIEW

File: release/train_tests/benchmark/dataloader_factory.py [hunk 1]
```diff
@@ -52,23 +56,27 @@ def collate_fn(self) -> Dict[str, Dataset]:
...
         ds_iterator = self._ray_ds_iterators["train"] = ray.train.get_dataset_shard(
             "train"
         )
         return iter(
             ds_iterator.iter_torch_batches(
-                batch_size=batch_size,
-                local_shuffle_buffer_size=batch_size * 8,
+                batch_size=self.config.train_batch_size,
+                local_shuffle_buffer_size=(
+                    self.config.local_buffer_shuffle_size
```
Comment (srinathk10) [inline release/train_tests/benchmark/dataloader_factory.py:67]: Nice!

**Review by srinathk10 — APPROVED** [review]
